### PR TITLE
feat(metrics): Emit additional data for the DataFlow fraud detection pipeline.

### LINKED ISSUE #2858: Emit additional metadata with events for consumption by the fraud detection pipeline
This is a placeholder to be filled in as we find out more from @ameihm0912 and @moz-jvehent about the additional metadata needed for the new fraud detection pipeline.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Adds `email`, `phoneNumber` and `method`

fixes #2858

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@vbudhram - r?

File: lib/log.js [hunk 1]
```diff
@@ -94,17 +97,18 @@ Lug.prototype.summary = function (request, response) {
...
     accountRecreated: request.app.accountRecreated,
-    t: Date.now() - request.info.received
+    t: Date.now() - request.info.received,
+    uid: credentials.uid || payload.uid || query.uid || response.uid || responseBody.uid || '00',
+    service: payload.service || query.service,
+    reason: payload.reason || query.reason,
+    redirectTo: payload.redirectTo || query.redirectTo,
+    keys: query.keys,
+
+    // Additional data used by the DataFlow fraud detection pipeline
```
Comment (vbudhram) [inline lib/log.js]: Should we mention anything about PII here and why it is ok?

File: test/local/log.js [hunk 2]
```diff
@@ -462,29 +462,137 @@ describe('log', () => {
...
+      auth: {
+        credentials: {
+          email: 'quix',
+          uid: 'quid',
+        }
+      },
       emitRouteFlowEvent: emitRouteFlowEvent,
-      headers: {},
+      headers: {
+        'user-agent': ''
```
Comment (vbudhram) [inline test/local/log.js]: Should we put a value in for header so we can test that it is not empty string in assert?

File: test/local/log.js [hunk 3]
```diff
@@ -462,29 +462,137 @@ describe('log', () => {
...
     assert.equal(logger.info.callCount, 1)
-    assert.equal(logger.info.args[0][1].op, 'request.summary')
-    assert.equal(logger.info.args[0][1].status, 201)
-    assert.equal(logger.info.args[0][1].code, undefined)
+
+    const line = logger.info.args[0][1]
+
+    // Because t is generated using Date.now and subtracting info.received,
+    // it should be >= 0, but we don't know the exact value.
+    assert.isNumber(line.t)
```
Comment (vbudhram) [inline test/local/log.js]: You could mock `Data.now` but meh.

File: test/local/log.js [hunk 4]
```diff
@@ -462,29 +462,137 @@ describe('log', () => {
...
+      lang: 'en',
+      agent:'',
+      remoteAddressChain: ['95.85.19.180', '78.144.14.50'],
+      accountRecreated: false,
+      uid: 'quid',
+      service: 'corge',
+      reason: 'grault',
+      redirectTo: 'garply',
+      keys: true,
+
```
Comment (vbudhram) [inline test/local/log.js]: Space here for sparating Dataflow values? If so, you could put the `method` value here too.

File: test/local/log.js (hunk 2 above)
Comment (shane-tomlinson) [reply test/local/log.js]: You are probably right.

File: lib/log.js (hunk 1 above)
Comment (shane-tomlinson) [reply lib/log.js]: Good idea.

**Pull request merged**